impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtParserBuilder.java

```diff
@@ -170,6 +172,14 @@ public JwtParserBuilder setCompressionCodecResolver(CompressionCodecResolver com
 
     @Override
     public JwtParser build() {
+
+        // Only lookup the deserializer IF it is null. It is possible a Deserializer implementation was set
+        // that is NOT exposed as a service and no other implementations are available for lookup.
+        if (this.deserializer == null) {
+            // try to find one based on the services available:
+            this.deserializer = Services.loadFirst(Deserializer.class);
```

Comment:
same

Reply:
This one is a new interface, so we don't need to retain a previous behavior